src/Database/test.js

```diff
@@ -102,21 +102,21 @@ describe('Batch writes', () => {
       ['update', m2],
     ])
 
-    expect(collectionObserver).toHaveBeenCalledTimes(4)
+    expect(collectionObserver).toHaveBeenCalledTimes(1)
     expect(collectionObserver).toHaveBeenCalledWith([
-      { record: m1, type: CollectionChangeTypes.updated },
-    ])
-    expect(collectionObserver).toHaveBeenCalledWith([
-      { record: m2, type: CollectionChangeTypes.updated },
+        { record: m3, type: CollectionChangeTypes.created },
+        { record: m1, type: CollectionChangeTypes.updated },
+        { record: m4, type: CollectionChangeTypes.created },
+        { record: m2, type: CollectionChangeTypes.updated },
     ])
 
     const createdRecords = [m3, m4]
     createdRecords.forEach(record => {
       expect(record._isCommitted).toBe(true)
       expect(collection._cache.get(record.id)).toBe(record)
-      expect(collectionObserver).toHaveBeenCalledWith([
-        { record, type: CollectionChangeTypes.created },
-      ])
+      // expect(collectionObserver).toHaveBeenCalledWith([
+        // { record, type: CollectionChangeTypes.created },
+      // ])
```

Comment:
Not sure how to replace this.

Reply:
just delete it — you're explicitly checking all calls on `collectionObserver` a few linea above and that should be enough 👍